filecoindot/src/ocw/mod.rs

```diff
@@ -31,31 +30,40 @@ pub const FILECOIN_RPC: &[u8] = b"FILECOIN_RPC";
 
 /// offchain worker entry
 pub fn offchain_worker<T: Config>(block_number: T::BlockNumber) -> Result<()> {
-    let url = StorageValueRef::persistent(FILECOIN_RPC)
+    // get encoded urls from storage
+    let urls = StorageValueRef::persistent(FILECOIN_RPC)
         .get::<Vec<u8>>()
         .map_err(|_| Error::GetStorageFailed)?
         .ok_or(Error::FilecoinRpcNotSet)?;
 
-    // log errors from ocw
-    bootstrap::<T>(
-        block_number,
-        core::str::from_utf8(&url).map_err(|_| Error::FormatBytesFailed)?,
-    )?;
+    // decode endpoints
+    let endpoints = core::str::from_utf8(&urls)
```

Comment:
urls can contain commas in the parameters query parts, a typical endpoints may not have one, but a user could really mess with this. I suggest to use the url crate and use array when passing the urls.